src/coreComponents/physicsSolvers/fluidFlow/kernels/singlePhase/ThermalHydrostaticPressureKernel.hpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2016-2024 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2024 TotalEnergies
+ * Copyright (c) 2018-2024 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2023-2024 Chevron
+ * Copyright (c) 2019-     GEOS/GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ThermalHydrostaticPressureKernel.hpp
+ */
+
+#ifndef GEOS_PHYSICSSOLVERS_FLUIDFLOW_SINGLEPHASE_THERMALHYDROSTATICPRESSUREKERNEL_HPP
+#define GEOS_PHYSICSSOLVERS_FLUIDFLOW_SINGLEPHASE_THERMALHYDROSTATICPRESSUREKERNEL_HPP
+
+#include "common/DataTypes.hpp"
+#include "common/GEOS_RAJA_Interface.hpp"
+#include "constitutive/fluid/singlefluid/SingleFluidBase.hpp"
+
+namespace geos
+{
+
+namespace thermalSinglePhaseBaseKernels
+{
+
+/******************************** HydrostaticPressureKernel ********************************/
+
+struct HydrostaticPressureKernel
+{
+
+  template< typename FLUID_WRAPPER >
+  static bool
+  computeHydrostaticPressure( integer const maxNumEquilIterations,
+                              real64 const & equilTolerance,
+                              real64 const (&gravVector)[ 3 ],
+                              FLUID_WRAPPER fluidWrapper,
+                              TableFunction::KernelWrapper tempTableWrapper,
```

Comment:
Shouldn't temperature value be passed as a `real64 const`/`real64` to be consistent with unitarian nature of the intended interface ?
```suggestion
                              real64 const & temp,
```